Pocket Casts Watch App/SourceInterfaceModel.swift

```diff
@@ -0,0 +1,176 @@
+import PocketCastsServer
+import PocketCastsUtils
+import WatchKit
+
+class SourceInterfaceModel: ObservableObject {
+
+    @Published var activeSource: Source = .phone
+
+    @Published var lastRefreshLabel: String = L10n.profileLastAppRefresh(L10n.timeFormatNever)
+
+    @Published var isPlusUser: Bool = false
+
+    @Published var isLoggedIn: Bool = false
+
+    @Published var profileImage: String = "profile-free"
+
+    @Published var usernameLabel: String = L10n.signedOut
+
+    private var refreshTimedActionHelper = TimedActionHelper()
+
+    func willActivate() {
+        addObservers()
+        reload()
+        handleDataUpdated()
+    }
+
+    func handleDataUpdated() {
+        guard !refreshTimedActionHelper.isTimerValid() else {
+            refreshTimedActionHelper.cancelTimer()
+            RefreshManager.shared.refreshPodcasts(forceEvenIfRefreshedRecently: false)
+            return
+        }
+        reload()
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: WatchConstants.Notifications.dataUpdated), object: nil)
+        removeAllCustomObservers()
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    // MARK: - Notifications For Updates
+
+    @objc private func dataDidUpdate() {
+        DispatchQueue.main.async {
+            self.handleDataUpdated()
+        }
+    }
+
+    private var customObservers = [Notification.Name]()
+
+    func addCustomObserver(_ name: Notification.Name, selector: Selector) {
+        if containsObserver(name) { return } // we already have this one
+
+        customObservers.append(name)
+
+        NotificationCenter.default.addObserver(self, selector: selector, name: name, object: nil)
+    }
+
+    func removeAllCustomObservers() {
+        if customObservers.count == 0 { return }
+
+        let notCenter = NotificationCenter.default
+        for name in customObservers {
+            notCenter.removeObserver(self, name: name, object: nil)
+        }
+        customObservers.removeAll()
+    }
+
+    private func containsObserver(_ name: Notification.Name) -> Bool {
+        if customObservers.count == 0 { return false }
+
+        return customObservers.contains(name)
+    }
+
+    func addObservers() {
+        addCustomObserver(NSNotification.Name(rawValue: WatchConstants.Notifications.dataUpdated), selector: #selector(dataDidUpdate))
+        addCustomObserver(Notification.Name(rawValue: WatchConstants.Notifications.loginStatusUpdated), selector: #selector(handleStatusChangeFromNotification))
+        addCustomObserver(Notification.Name(rawValue: ServerNotifications.subscriptionStatusChanged.rawValue), selector: #selector(handleStatusChangeFromNotification))
```

Comment:
I believe these are already NotificationNames so can't they be changed to remove the wrapper?

```suggestion
        addCustomObserver(ServerNotifications.subscriptionStatusChanged, selector: #selector(handleStatusChangeFromNotification))
```

Reply:
Good catch, I updated it here: https://github.com/Automattic/pocket-casts-ios/pull/1800/commits/c541ffdbb22b9273ca86b36fe8c2a80bb116a5f7